pkg/api/http/http_test.go

```diff
@@ -4682,33 +4665,29 @@ func TestStateStoreErrors(t *testing.T) {
 
 func TestExtractEtag(t *testing.T) {
 	t.Run("no etag present", func(t *testing.T) {
-		r := fasthttp.RequestCtx{
-			Request: fasthttp.Request{},
-		}
-
-		ok, etag := extractEtag(&r)
+		r, err := nethttp.NewRequest("GET", "http://localhost", nil)
+		require.NoError(t, err)
+		ok, etag := extractEtag(r)
 		assert.False(t, ok)
 		assert.Empty(t, etag)
 	})
 
 	t.Run("empty etag exists", func(t *testing.T) {
-		r := fasthttp.RequestCtx{
-			Request: fasthttp.Request{},
-		}
-		r.Request.Header.Add("If-Match", "")
+		r, err := nethttp.NewRequest("GET", "http://localhost", nil)
```

Comment:
for int tests we use 127.0.0.1 to prevent any weirdness/flakiness, I assume we should do the same here?

```suggestion
		r, err := nethttp.NewRequest("GET", "http://127.0.0.1", nil)
```